replica/database.cc

```diff
@@ -2054,29 +2026,65 @@ future<> database::apply_in_memory(const mutation& m, column_family& cf, db::rp_
     return cf.apply(m, std::move(h), timeout);
 }
 
-future<mutation> database::apply_counter_update(schema_ptr s, const frozen_mutation& m, db::timeout_clock::time_point timeout, tracing::trace_state_ptr trace_state) {
+future<counter_update_guard> database::acquire_counter_locks(schema_ptr s, const frozen_mutation& fm, db::timeout_clock::time_point timeout, tracing::trace_state_ptr trace_state) {
+    auto& cf = find_column_family(fm.column_family_id());
+
+    auto m = fm.unfreeze(s);
+    m.upgrade(cf.schema());
+
+    auto op = cf.write_in_progress();
+
+    tracing::trace(trace_state, "Acquiring counter locks");
+
+    co_return co_await update_write_metrics([&m, &cf, op = std::move(op), timeout] mutable -> future<counter_update_guard> {
```

Comment:
`update_write_metrics` doesn't measure the write operation, we now increment `total_writes` three times for a single counter write -- in `acquire_counter_locks`, `prepare_counter_update` and `apply_counter_update`.